src/main/java/io/fabric8/maven/docker/service/RunService.java

```diff
@@ -71,7 +72,9 @@
  * @since 16/06/15
  */
 public class RunService {
-
+    private static final int UPDATE_MAPPED_PORTS_RETRIES = 5;
+    private static final int UPDATE_MAPPED_PORTS_RETRY_DELAY_MS = 500;
```

Comment:
i would recommend to go here to 250 and increase the UPDATE_MAPPED_PORTS_RETRIES

Reply:
i checked the logs with this parameters is see only one try per container. i will test it with a lower time
[WARNING] DOCKER> Failed to get port bindings for container 4d2b4ff6615c (attempt 0 of 5)
[WARNING] DOCKER> Failed to get port bindings for container 35efb753ef46 (attempt 0 of 5)

Reply:
private static final int UPDATE_MAPPED_PORTS_RETRIES = 50;
    private static final int UPDATE_MAPPED_PORTS_RETRY_DELAY_MS = 50;
    
    i still see only one attemt
    [WARNING] DOCKER> Failed to get port bindings for container 8c64d1409bef (attempt 0 of 50)

Reply:
private static final int UPDATE_MAPPED_PORTS_RETRIES = 500;
    private static final int UPDATE_MAPPED_PORTS_RETRY_DELAY_MS = 5;
    
    [WARNING] DOCKER> Failed to get port bindings for container 0fbb04f25568 (attempt 0 of 500)
    [WARNING] DOCKER> Failed to get port bindings for container 0fbb04f25568 (attempt 1 of 500)

note that a have a pretty new PC with a lot of power.

so finally i would recommend 10ms

Reply:
Yeah, i think the issue is it only prints out when it fails (also it is 0 indexed) so it may fail once and then pass on the 2nd attempt but will not print anything out. Will tweak this.

Reply:
I think having the delay of 250ms and 10 attempts is reasonable- it really should not make a huge difference to container start times to wait 250ms between attempts, especially as it failing to get the port data from docker is an edge case and race condition. As container start times are usually in the order of seconds or tens of seconds.

Reply:
I would change this
`private static final int UPDATE_MAPPED_PORTS_RETRY_DELAY_MS = 500;`
to
`private static final int UPDATE_MAPPED_PORTS_RETRY_DELAY_MS = 10;`
and also adjust the UPDATE_MAPPED_PORTS_RETRIES to match again.

Reply:
Im not convinced that hammering the docker api every 10ms for (potentially) up to 500 attempts is the best idea. By setting the delay to 250ms it will wait long enough that the port information should be available after 1 retry. 250ms should not be problematic to add to container start times.

Happy to be told otherwise by the maintainers of this codebase but IMO 250ms and 10 attempts should be sufficient.